File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EPrimaryFixedDimLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EPrimaryFixedDimLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl21
            AppTheme.STANDARD -> R.color.cstl21
            AppTheme.BLUE -> R.color.cbl21
            AppTheme.GREEN -> R.color.cgl21
            AppTheme.RED -> R.color.crl21
            AppTheme.NOTHING -> R.color.cnol21
        }
    }
}

File: lib/layer2_core/light/e_primary_fixed_dim_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EPrimaryFixedDimLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl21;
      case 'standard': return cstl21;
      case 'blue': return cbl21;
      case 'green': return cgl21;
      case 'red': return crl21;
      case 'nothing': return cnol21;
      default: return cstl21;
    }
  }
}
